### PR TITLE
LRAC-8156 Adding CurrentACPlanLevelShownInCard Test

### PR DESCRIPTION
**Pull request by AfonsoHSantos**
https://issues.liferay.com/browse/LRAC-8156

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro [hunk 1]
```diff
@@ -40,6 +40,16 @@ definition {
 			value1 = "${pageViewsUsageValue}");
 	}
 
+	macro assertPlanDetails {
+		key_planName = "${planName}";
```
Comment (alexcandidos) [inline portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro:44]: Hi @AfonsoHSantos , could you please declare the variable: 
`var key_planName = "${planName}";`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/Usage.testcase [hunk 2]
```diff
@@ -39,12 +39,19 @@ definition {
...
+		ACUtils.launchAC();
+
+		ACProperties.switchProperty(propertyName = "${propertyName}");
+
+		ACNavigation.goToSettings();
 
-		// TODO LRAC-8156 CurrentACPlanLevelShownInCard pending implementation
+		ACSettings.goToUsage();
 
+		ACSettings.assertPlanDetails(planName = "Enterprise");
```
Comment (alexcandidos) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/Usage.testcase:54]: Hey @AfonsoHSantos , I would suggest you to pass the values you want to validate regarding the plan, for example:
- planName = "Enterprise";
- individualsLimit = "100,000";
- pageViews = "60,000,000";
- usersLimit = "Unlimited";
- individualsAddOns = "5,000";
- pageViewsAddOns = "5,000,000";
- individualsAddOnsPrice = "500 USD";
- pageViewsAddOnsPrice = "250 USD";

So, you will validate these values on this assertPlanDetails macro.

File: portal-web/test/functional/com/liferay/portalweb/paths/analyticscloud/Usage.path [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+</tr>
+<tr>
+	<td>USAGE_PLAN_LIMITS_TABLE</td>
+	<td>//div[@class='limits']//ul//li[1]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>USAGE_PLAN_ADDONS_TABLE</td>
+	<td>//div[@class='addons']//ul//li[1]</td>
+	<td></td>
```
Comment (alexcandidos) [inline portal-web/test/functional/com/liferay/portalweb/paths/analyticscloud/Usage.path:26]: Hi @AfonsoHSantos , I would suggest you update these paths to validate the values.

**Review by alexcandidos — CHANGES_REQUESTED** [review]
Hi @AfonsoHSantos , I left some suggestions to you :)

**Comment by AfonsoHSantos** [comment]
fixed.

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro [hunk 4]
```diff
@@ -40,6 +40,64 @@ definition {
 			value1 = "${pageViewsUsageValue}");
 	}
 
+	macro assertPlanDetails {
```
Comment (alexcandidos) [inline portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro:43]: Hi @AfonsoHSantos , it seems the macro ``asserPlanDetails`` is duplicated. Could you please double-check?

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro (hunk 1 above)
Comment (AfonsoHSantos) [reply portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro]: fixed

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/Usage.testcase (hunk 2 above)
Comment (AfonsoHSantos) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/Usage.testcase]: fixed

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro (hunk 4 above)
Comment (AfonsoHSantos) [reply portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro:43]: fixed

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro [hunk 5]
```diff
@@ -40,6 +40,35 @@ definition {
...
+			var planItem = StringUtil.extractFirst("${limits}","-");
+			var planLimits = StringUtil.extractLast("${limits}","-");
+			var tableName = "limits";
+
+			AssertElementPresent(
+				key_planItem = "${planItem}",
+				key_planLimits = "${planLimits}",
+				key_tableName = "${tableName}",
+				locator1 = "ACUsage#USAGE_PLAN_LIMITS_TABLE");
+		}
```
Comment (alexcandidos) [inline portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro:57]: Hi @AfonsoHSantos , what do you think about improving the code legibility by removing the ``for`` loops and the String manipulation... I thought in something like this:
```

AssertElementPresent(
				key_planItem = "Individuals",
				key_planLimits = "${individualsLimit}",
				key_tableName = "limits",
				locator1 = "ACUsage#USAGE_PLAN_LIMITS_TABLE");

AssertElementPresent(
				key_planItem = "Page Views",
				key_planLimits = "${pageViews}",
				key_tableName = "limits",
				locator1 = "ACUsage#USAGE_PLAN_LIMITS_TABLE");

AssertElementPresent(
				key_planItem = "Users",
				key_planLimits = "${usersLimit}",
				key_tableName = "limits",
				locator1 = "ACUsage#USAGE_PLAN_LIMITS_TABLE");
```

And the same breakdown logic for the addon's section.

**Comment by ViniLopes20** [comment]
Looks good to me!

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro (hunk 5 above)
Comment (r4ddek) [reply portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro:57]: +1

Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro:57]: +1

Comment (AfonsoHSantos) [reply portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro]: fixed

**Comment by alexcandidos** [comment]
Hi @AfonsoHSantos , Thanks for working on it!

The code looks good to me!

**Comment by alexcandidos** [comment]
ci:forward

**Pull request closed without merging**